### PR TITLE
1.21.2/1.21.3

### PR DESCRIPTION
**Pull request by AzureAaron**
Beyond supporting 1.21.2/1.21.3 this also changes:
- Rewrite the recipe book implementation from the ground up
- Make Enderman Slayer beacon highlighter work more reliably incase the block update is sent via a chunk delta update
- Fix the Skyblocker screen drawing the blur background twice
- Repackages the chat filters from `skyblock.filters` to `skyblock.chat.filters`
- Removes now unnecessary Sodium reflection, Vanilla render thread scheduling is now always used
- Adds profiling capabilities for the Translucent Draw & Scheduler Tick.
- Fixes there being a gap for the status effects overlay with REI when you disable the overlay through Skyblocker

### CODE REVIEW

File: src/main/java/de/hysky/skyblocker/skyblock/events/EventToast.java [hunk 1]
```diff
@@ -90,4 +91,14 @@ protected int getInnerContentsHeight() {
     public int getHeight() {
         return Math.max(getInnerContentsHeight() + 12 + 2, 32);
     }
+
+	@Override
+	public Visibility getVisibility() {
+		return toastTime > 5_000 ? Visibility.HIDE: Visibility.SHOW;
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/skyblock/events/EventToast.java:97]: nit:
```suggestion
		return toastTime > 5_000 ? Visibility.HIDE : Visibility.SHOW;
```

File: src/main/java/de/hysky/skyblocker/skyblock/itemlist/recipebook/NoopRecipeScreenHandler.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import net.minecraft.screen.ScreenHandlerType;
+import net.minecraft.server.world.ServerWorld;
+
+/**
+ * Implementation of an {@link AbstractRecipeScreenHandler} that does not do anything.
+ */
+public class NoopRecipeScreenHandler extends AbstractRecipeScreenHandler {
+
+	protected NoopRecipeScreenHandler() {
+		super(ScreenHandlerType.GENERIC_9X6, -88);
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/skyblock/itemlist/recipebook/NoopRecipeScreenHandler.java:19]: Anything specific about this sync id?

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 3]
```diff
@@ -35,15 +35,15 @@ body:
       label: Minecraft Version
       description: What version of Minecraft are you running? If you do not know what version you are using, look in the bottom left corner of the main menu in game.
       options:
-        - "1.21.0/1.21.1"
+        - "1.21.2/1.21.3"
```
[the remark this answers is not in the document]
Comment (AzureAaron) [reply .github/ISSUE_TEMPLATE/bug_report.yml:38]: I tried to leave them but I got a YAML syntax error and gave up

File: src/main/java/de/hysky/skyblocker/skyblock/itemlist/recipebook/NoopRecipeScreenHandler.java (hunk 2 above)
Comment (AzureAaron) [reply src/main/java/de/hysky/skyblocker/skyblock/itemlist/recipebook/NoopRecipeScreenHandler.java:19]: 8 is a lucky number!

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (AzureAaron) [reply .github/ISSUE_TEMPLATE/bug_report.yml:39]: I figured it out, it inserted tabs rather than spaces (dunno why YAML hates tabs)

**Review by kevinthegreat1 — APPROVED** [review]

**Pull request merged**